core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1396,6 +1398,8 @@ class ReplicaManager(val config: KafkaConfig,
                     stateChangeLogger.info(s"Updating log for $topicPartition to assign topic ID " +
                       s"$topicId from LeaderAndIsr request from controller $controllerId with correlation " +
                       s"id $correlationId epoch $controllerEpoch")
+                    if (partitionState.leader != localBrokerId && metadataCache.hasAliveBroker(partitionState.leader))
```

Comment:
nit: I think that I would prefer to have `metadataCache.hasAliveBroker(partitionState.leader` in  `updateTopicIdForFollowers` after all. We could do this check when we build the Map at L1778.

Reply:
Don't forget to remove `&& metadataCache.hasAliveBroker(partitionState.leader)` here.